modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuilder.java

```diff
@@ -133,71 +176,77 @@ private BuildIndexTask(
 
         @Override
         public void run() {
-            if (!busyLock.enterBusy()) {
-                return;
-            }
+            completedFuture(null)
```

Comment:
The first action in the `thenCompose` callback creates a future, do we really need `completedFuture(null)` here? 😰

Reply:
Conveniently, we will always execute `whenComplete` below (L243), with no additional code in `finally` block to do the same.